js/subway-main.js

```diff
@@ -305,29 +342,30 @@ $(function() {
             $('#custom-line-error').text('Enter a name.');
             issue = true;
         }
-        if (find_line_by_name(custom_line_name) != null) {
-            $('#custom-line-name').addClass('issue');
-            $('#custom-line-error').text('Name already in use.');
-            issue = true;
-        }
-        if (!is_css_color(custom_line_css_bg)) {
-            $('#custom-line-css-bg').addClass('issue');
-            $('#custom-line-error').text('Use a valid hex color code.');
-            issue = true;
-        }
-        if (!is_css_color(custom_line_css_text)) {
-            $('#custom-line-css-text').addClass('issue');
-            $('#custom-line-error').text('Use a valid hex color code.');
-            issue = true;
-        }
-        var custom_line_css_class = 'subway-line';
-        if ($("#custom-line-marker").hasClass('subway-line-long')) {
-            custom_line_css_class = 'subway-line-long';
-        }
-        if (!issue) {
+        if (find_line_by_name(custom_line_name) == null) {
```

Comment:
It looks like the effect of this change is that you can create a new line with the same name as an existing line, which only has the effect of potentially changing the line's color, but the UI is not updated and it does not persist across saves.
Also, by not checking for `!issue`, it's possible to create a line with no color, just showing up as an invisible line.

Reply:
Put the if(!issue) back in in a judicious place